test/e2e/dependenciesdistributor_test.go

```diff
@@ -0,0 +1,136 @@
+					func(deployment *appsv1.Deployment) bool {
+						return true
+					})
+
+				framework.WaitSecretPresentOnClustersFitWith(initClusterNames, secret.Namespace, secretName,
+					func(secret *corev1.Secret) bool {
+						return true
+					})
+
+				framework.WaitConfigMapPresentOnClustersFitWith(initClusterNames, configMap.Namespace, configMapName,
+					func(configmap *corev1.ConfigMap) bool {
+						return true
+					})
+			})
+
+			ginkgo.By("updating propagation policy's clusterNames", func() {
+				patch := []map[string]interface{}{
+					{
+						"op":    "replace",
+						"path":  "/spec/placement/clusterAffinity/clusterNames",
+						"value": updateClusterNames,
+					},
+				}
+
+				framework.PatchPropagationPolicy(karmadaClient, policy.Namespace, policyName, patch, types.JSONPatchType)
+				framework.WaitDeploymentPresentOnClustersFitWith(updateClusterNames, deployment.Namespace, deploymentName,
+					func(deployment *appsv1.Deployment) bool {
+						return true
+					})
+
+				framework.WaitSecretPresentOnClustersFitWith(updateClusterNames, secret.Namespace, secretName,
+					func(secret *corev1.Secret) bool {
+						return true
+					})
+
+				framework.WaitConfigMapPresentOnClustersFitWith(updateClusterNames, configMap.Namespace, configMapName,
+					func(configmap *corev1.ConfigMap) bool {
+						return true
+					})
+			})
+
+			ginkgo.By("updating deployment's volumes", func() {
+				updateVolumes := []corev1.Volume{
+					{
+						Name: "vol-configmap",
+						VolumeSource: corev1.VolumeSource{
+							ConfigMap: &corev1.ConfigMapVolumeSource{
+								LocalObjectReference: corev1.LocalObjectReference{
+									Name: configMapName,
+								},
+							},
+						},
+					},
+				}
+
+				framework.UpdateDeploymentVolumes(kubeClient, deployment, updateVolumes)
+				framework.WaitSecretDisappearOnClusters(updateClusterNames, secret.Namespace, secretName)
+			})
+
+			ginkgo.By("updating configmap's data", func() {
+				patch := []map[string]interface{}{
+					{
+						"op":    "replace",
+						"path":  "/data/user",
+						"value": "karmada-e2e",
+					},
+				}
+
+				framework.UpdateConfigMapWithPatch(kubeClient, configMap.Namespace, configMapName, patch, types.JSONPatchType)
+				framework.WaitConfigMapPresentOnClustersFitWith(updateClusterNames, configMap.Namespace, configMapName,
+					func(configmap *corev1.ConfigMap) bool {
+						for key, value := range configmap.Data {
+							if key == "user" && value == "karmada-e2e" {
+								return true
+							}
+						}
+						return false
+					})
+			})
+
+			framework.RemoveDeployment(kubeClient, deployment.Namespace, deployment.Name)
+			framework.WaitDeploymentDisappearOnClusters(framework.ClusterNames(), deployment.Namespace, deployment.Name)
+			framework.RemovePropagationPolicy(karmadaClient, policy.Namespace, policy.Name)
+			framework.RemoveSecret(kubeClient, secret.Namespace, secretName)
+			framework.RemoveConfigMap(kubeClient, configMap.Namespace, configMapName)
```

Comment:
This part can be placed in `ginkgo.AfterEach` to avoid resource residue caused by errors in the previous steps:
```diff
--- a/test/e2e/dependenciesdistributor_test.go
+++ b/test/e2e/dependenciesdistributor_test.go
@@ -38,6 +38,14 @@ var _ = ginkgo.Describe("[DependenciesDistributor] automatically propagate relev
                secret := testhelper.NewSecret(testNamespace, secretName, map[string][]byte{"user": []byte("karmada")})
                configMap := testhelper.NewConfigMap(testNamespace, configMapName, map[string]string{"user": "karmada"})

+               ginkgo.AfterEach(func() {
+                       framework.RemoveDeployment(kubeClient, deployment.Namespace, deployment.Name)
+                       framework.WaitDeploymentDisappearOnClusters(framework.ClusterNames(), deployment.Namespace, deployment.Name)
+                       framework.RemovePropagationPolicy(karmadaClient, policy.Namespace, policy.Name)
+                       framework.RemoveSecret(kubeClient, secret.Namespace, secretName)
+                       framework.RemoveConfigMap(kubeClient, configMap.Namespace, configMapName)
+               })
+
                ginkgo.It("dependencies automatically propagation testing", func() {
                        framework.CreatePropagationPolicy(karmadaClient, policy)
                        framework.CreateDeployment(kubeClient, deployment)
@@ -125,12 +133,6 @@ var _ = ginkgo.Describe("[DependenciesDistributor] automatically propagate relev
                                                return false
                                        })
-
-                       framework.RemoveDeployment(kubeClient, deployment.Namespace, deployment.Name)
-                       framework.WaitDeploymentDisappearOnClusters(framework.ClusterNames(), deployment.Namespace, deployment.Name)
-                       framework.RemovePropagationPolicy(karmadaClient, policy.Namespace, policy.Name)
-                       framework.RemoveSecret(kubeClient, secret.Namespace, secretName)
-                       framework.RemoveConfigMap(kubeClient, configMap.Namespace, configMapName)
                })
```
This problem also exists in other original e2e use cases, I will modify it later.